src/org/thoughtcrime/securesms/notifications/MessageNotifier.java

```
@@ -420,11 +424,28 @@ private static void setNotificationAlarms(Context context,
     String ledBlinkPattern       = TextSecurePreferences.getNotificationLedPattern(context);
     String ledBlinkPatternCustom = TextSecurePreferences.getNotificationLedPatternCustom(context);
     String[] blinkPatternArray   = parseBlinkPattern(ledBlinkPattern, ledBlinkPatternCustom);
+    Uri ringtoneUri              = TextUtils.isEmpty(ringtone) || !signal ? null : Uri.parse(ringtone);
+    TelephonyManager telephony   = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
 
-    builder.setSound(TextUtils.isEmpty(ringtone) || !signal ? null : Uri.parse(ringtone));
+    if (telephony.getCallState() == TelephonyManager.CALL_STATE_IDLE) {
+      builder.setSound(ringtoneUri);
 
-    if (signal && vibrate) {
-      builder.setDefaults(Notification.DEFAULT_VIBRATE);
+      if (signal && vibrate) {
+        builder.setDefaults(Notification.DEFAULT_VIBRATE);
+      }
+    }
+    else {
+      AudioManager audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
+      if (audioManager.getRingerMode() == AudioManager.RINGER_MODE_NORMAL && ringtoneUri != null) {
+        Ringtone tone = RingtoneManager.getRingtone(context, Uri.parse(ringtone));
+        tone.setStreamType(AudioManager.STREAM_VOICE_CALL);
+        tone.play();
+      }
+
+      if (signal && vibrate) {
+        Vibrator vibrator = (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
+        vibrator.vibrate(new long[] {0, 250, 250, 250}, -1);
```

default found in `NotificationManagerService` source, class is `@hide`n